internal/util/addmergecomment/addmergecomment.go

```diff
@@ -49,6 +62,35 @@ func Process(paths ...string) error {
 	return nil
 }
 
+// addUpstreamAnnotation adds internal.kpt.dev/upstream-identifier annotation to resource.
+func addUpstreamAnnotation(object *kyaml.RNode, mergeComment string) error {
+	annotations := object.GetAnnotations()
+	if _, ok := annotations[upstreamIdentifier]; ok {
+		// skip adding annotation if the upstream annotation is already present
```

Comment:
If I clone a package that already has these annotations (that is a case for any 3 level pkg chain), this will skip adding annotation ? with this, all descendant packages will have GVKNN of the resources in the root pkg.

Reply:
I changed the code to use immediate upstream meta as the origin. Also added a comment to mark the potential caveats in 3-way-merge